domain/src/main/java/joist/domain/uow/UnitOfWork.java

```diff
@@ -137,6 +129,24 @@ void setUpdater(Updater updater) {
     }
   }
 
+  <T extends DomainObject> List<T> load(Class<T> type, Collection<Long> ids) {
+    boolean containsAllIds = this.identityMap.getIdsOf(type).containsAll(ids);
+    if (containsAllIds) {
+      List<T> loaded = new ArrayList<T>();
+      for (Long id : ids) {
+        loaded.add((T) this.identityMap.findOrNull(type, id));
+      }
+      return loaded;
+    }
+
+    Alias<T> a = AliasRegistry.get(type);
+    try {
+      return Select.from(a).where(a.getIdColumn().in(ids)).list();
```

Comment:
You could potentially subtract the ids we've already loaded, e.g. if they ask for [1, 2, 3], and we already have id 2 in the IdentityMap, we could query for just 1, 3.